src/main/java/seedu/address/model/session/Interval.java

```diff
@@ -52,25 +53,56 @@ public LocalDateTime getStart() {
         return this.start;
     }
 
+    public LocalDateTime getEnd() {
+        return this.start.plusMinutes(durationInMinutes);
+    }
+
     public String getFormattedStartDateTime(DateTimeFormatter formatter) {
-        return this.start.format(formatter);
+        return this.getStart().format(formatter);
     }
 
-    public LocalDateTime getEnd() {
-        return this.start.plusMinutes(durationInMinutes);
+    public String getFormattedEndDateTime(DateTimeFormatter formatter) {
+        return this.getEnd().format(formatter);
+    }
+
+    /**
+     * Returns a String representation of start and end time depending on whether they are on different days.
+     * If same day, the String representation will specify the time of start and end.
+     * Else, also specify the date of the end time.
+     *
+     * @return String representation of start and end time specifying end date if interval is over >1 days.
+     */
+    public String getAdjustedStartDateTime() {
+        if (this.isPastMidnight()) {
+            return getFormattedStartDateTime(TIME_12HR_PATTERN_FORMATTER) + " - "
+                    + getFormattedEndDateTime(SIMPLE_DATE_TIME_PATTERN_FORMATTER) + " "
+                    + getFormattedEndDateTime(TIME_12HR_PATTERN_FORMATTER);
+        } else {
+            return getFormattedStartDateTime(TIME_12HR_PATTERN_FORMATTER) + " - "
+                    + getFormattedEndDateTime(TIME_12HR_PATTERN_FORMATTER);
+        }
+    }
+
+    /**
+     * Returns true if the start and end time are on different days.
+     * @return Boolean value of whether start and end time are on different days
+     */
+    public boolean isPastMidnight() {
+        assert(this.getStart() != null && this.getEnd() != null);
+        return this.getEnd().truncatedTo(DAYS).isAfter(this.getStart().truncatedTo(DAYS));
```

Comment:
What about 10 Feb - 9 Mar? Will it show the date?

Reply:
Yup it'll show the date correctly! Bcos truncated returns a new LocalDateTime, not a int, so isBefore will work correctly. 

Just realised a issue though. What if I set start as 10 Feb 2017 and duration as a year. It'll display the start time and end time as the same.

Do I need another isPastNewYear? ... Or should I just change it to DDMMYYYY

Reply:
Nevermind, I think I'll change add YYYY to any interval in the session time. Sat 13 Feb -> Sat 13 Feb 2020

Reply:
Yea I guess for > 1 days, just display the full date or (can apply this to <= 1 day too, but this might clutter the screen tho)

Reply:
For the time field: Within a day, stay as 3.00pm - 4.00pm. Over 2 days, add year (e.g. 3.00pm to 01 Sep 2020 4.00pm)
For the header of each session card, I'm adding year regardless of duration le, because right now 18 Feb 2020 and 18 Feb 2021 both show as 18 Feb. Though its sorted, let's play safe hahah